src/Simulator/IncrementalRedstoneSimulator/RedstoneSimulatorChunkData.h

```diff
@@ -0,0 +1,70 @@
+
+#pragma once
+
+#include "Vector3.h"
+#include "RedstoneHandler.h"
+#include "../RedstoneSimulator.h"
+#include <unordered_map>
+
+
+
+
+
+class cIncrementalRedstoneSimulatorChunkData : public cRedstoneSimulatorChunkData
+{
+
+public:
+	void WakeUp(const Vector3i & a_Position)
+	{
+		m_ActiveBlocks.push_back(a_Position);
+	}
+
+	cVector3iArray & GetActiveBlocks()
+	{ 
+		return m_ActiveBlocks; 
+	}
+
+	const cRedstoneHandler::PoweringData GetCachedPowerData(const Vector3i & a_Position) const
+	{
+		auto Result = m_CachedPowerLevels.find(a_Position);
+		return (Result == m_CachedPowerLevels.end()) ? cRedstoneHandler::PoweringData() : Result->second;
+	}
+
+	void SetCachedPowerData(const Vector3i & a_Position, cRedstoneHandler::PoweringData a_PoweringData)
+	{
+		m_CachedPowerLevels[a_Position] = a_PoweringData;
+	}
+
+	std::tuple<int, int, bool> * GetMechanismDelayInfo(const Vector3i & a_Position)
+	{
+		auto Result = m_MechanismDelays.find(a_Position);
+		return (Result == m_MechanismDelays.end()) ? nullptr : &Result->second;
```

Comment:
So you're using a raw pointer as an optional return. We really need an optional type for situations like this.